src/Npgsql/HardCancellationBlock.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Threading;
+
+namespace Npgsql
+{
+    readonly struct HardCancellationBlock : IDisposable
+    {
+        static readonly AsyncLocal<bool> _entered = new AsyncLocal<bool>();
```

Comment:
Statics start with `s_` and thread statics with `t_`. It's a common convention in .NET.

```suggestion
        static readonly AsyncLocal<bool> s_entered = new AsyncLocal<bool>();
```